src/SonarScanner.MSBuild.Common/ProcessRunner.cs

```diff
@@ -149,35 +149,24 @@ private void SetEnvironmentVariables(ProcessStartInfo psi, IDictionary<string, s
         }
     }
 
-    private void OnOutputDataReceived(DataReceivedEventArgs e, bool logOutput, TextWriter standardOutputWriter)
+    private void HandleProcessOutput(string data, bool stdOut, bool logOutput, TextWriter outputWriter, OutputToLogMessage outputToLogMessage)
     {
-        if (e.Data is not null)
+        if (data is not null
+            && outputToLogMessage?.Invoke(stdOut, data) is { } logMessage
+            && logMessage.Message.RedactSensitiveData() is { } redactedMsg)
         {
-            var redactedMsg = e.Data.RedactSensitiveData();
             if (logOutput)
             {
-                // It's important to log this as an important message because
-                // this the log redirection pipeline of the child process
-                logger.LogInfo(redactedMsg);
+                Action<string, object[]> logMethod = logMessage.Level switch
+                {
+                    LogLevel.Info => logger.LogInfo,
+                    LogLevel.Warning => logger.LogWarning,
+                    LogLevel.Error => logger.LogError,
+                    _ => static (_, _) => { },
+                };
+                logMethod(redactedMsg, []);
```

Comment:
I feel using an action here is overkill.
Is there a reason beyond saving 6 lines not to do
```suggestion
switch (logMessage.Level)
{
    case LogLevel.Info:
        logger.LogInfo(redactedMsg);
        break;
    case LogLevel.Warning:
        logger.LogWarning(redactedMsg);
        break;
    case LogLevel.Error:
        logger.LogError(redactedMsg);
        break;
    default:
        break;
}
```